src/main/java/duke/TaskList.java

```diff
@@ -0,0 +1,219 @@
+            indexes.add(ind);
+        };
+        this.consume(indStr, mark);
+        this.save();
+        if (indexes.size() == 0) {
+            throw new IllegalArgumentException("No tasks found.");
+        } else if (indexes.size() == 1) {
+            Task task = this.tasks.get(indexes.get(0));
+            return new String[]{ (task.isDone() ? "marked: " : "unmarked: ") + addOrdinal(indexes.get(0), task) };
+        } else {
+            List<String> outputs = indexes.stream().map(i -> "\t" + addOrdinal(i, this.tasks.get(i))).collect(Collectors.toList());
+            outputs.add(0, "marked:");
+            return outputs.toArray(new String[0]);
+        }
+    }
+
+    /**
+     * Deletes the task(s) at the given index(es).
+     *
+     * @param indStr the index(es) of the task(s) to be deleted.
+     * @return a string representation of the task(s) that were deleted.
+     */
+    public String[] delete(String[] indStr) {
+        List<Integer> indexes = new ArrayList<>();
+        Consumer<Integer> delete = ind -> {
+            if (this.tasks.size() <= ind) {
+                throw new IllegalArgumentException("Task not found: " + ind);
+            }
+            indexes.add(ind);
+        };
+        this.consume(indStr, delete);
+        if (indexes.size() == 0) {
+            throw new IllegalArgumentException("No tasks found.");
+        } else if (indexes.size() == 1) {
+            Task rmTask = this.tasks.remove((int) indexes.get(0));
+            this.save();
+            return new String[]{ "deleted: " + addOrdinal(indexes.get(0), rmTask) };
+        } else {
+            indexes.sort(Collections.reverseOrder());
+            List<String> outputs = new ArrayList<>();
+            for (int i : indexes) {
+                outputs.add("\t" + addOrdinal(i, this.tasks.remove(i)));
+            }
+            this.save();
+            Collections.reverse(outputs);
+            outputs.add(0, "deleted:");
+            return outputs.toArray(new String[0]);
+        }
+    }
+
+    private void consume(String[] argument, Consumer<Integer> consumer) {
+        try {
+            String[] inds = argument[0].split("\\s");
+            for (String s : inds) {
+                int ind = Integer.parseInt(s) - 1;
+                consumer.accept(ind);
+            }
+        } catch (NumberFormatException | IndexOutOfBoundsException e) {
+            throw new IllegalArgumentException("Invalid index.");
+        }
+    }
+
+    /**
+     * Performs a search for partial matches on task descriptions.
+     *
+     * @param argument the search term.
+     * @return a string representation of the tasks that matched the search term.
+     */
+    public String[] find(String[] argument) {
+        String keyword = argument[0].toLowerCase();
+        List<Integer> matchIndexes = IntStream.range(0, this.tasks.size()).parallel().filter(i -> this.tasks.get(i).getDesc().toLowerCase().contains(keyword)).boxed().toList();
```

Comment:
A small issue but it might be more readable if all the functions are not in 1 line